core/src/main/spark-2.3/org/apache/spark/sql/catalyst/catalog/TiDirectExternalCatalog.scala

```diff
@@ -38,7 +39,8 @@ class TiDirectExternalCatalog(tiContext: TiContext) extends ExternalCatalog {
 
   override def getTable(db: String, table: String): CatalogTable = {
     val schema = TiUtil.getSchemaFromTable(
-      meta.getTable(db, table).getOrElse(throw new NoSuchTableException(db, table))
+      meta.getTable(db, table).getOrElse(throw new NoSuchTableException(db, table)),
+      tiContext.conf.getInt(TiConfigConst.TYPE_SYSTEM_VERSION, 0)
```

Comment:
If there are multiple places with default version 0, there should be an constant for default version. Also in TiDirectExternalCatalog.scala